api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -286,6 +286,16 @@ const (
 
 	// ManagementPlatformAnnotation specifies the infrastructure platform of the underlying management cluster
 	ManagementPlatformAnnotation = "hypershift.openshift.io/management-platform"
+
+	// MachineHealthCheckTimeoutAnnotation allows overriding the default machine health check timeout for
+	// nodepools. The annotation can be set in either the HostedCluster or the NodePool. If set on both, the
+	// one on the NodePool takes precedence. The value is a go duration string with a number and a unit (ie. 8m, 1h, etc)
+	MachineHealthCheckTimeoutAnnotation = "hypershift.openshift.io/machine-health-check-timeout"
```

Comment:
Let's create a follow up jira to decide how to expose MHC in the NodePool API
https://issues.redhat.com/browse/HOSTEDCP-1680